### PR TITLE
[NPU][diffusion] npu support enable_torch_compile for torchair backend on diffusion models 

### PR DESCRIPTION
**Pull request by Alisehen**
## Motivation

support torch.compile for npu

## Modifications

Explicitly use the `torchair` backend for `torch.compile` on the NPU path in the denoising stage.

## Benchmarking and Profiling

910B, Qwen-image

```
# base: 23.38s
sglang generate \
  --model-path=/model/qwen-image/ \
  --prompt='A fantasy landscape with mountains and a river, detailed, vibrant colors' \
  --width=1024 \
  --height=1024 \
  --vae-precision bf16 \
  --num-inference-steps=20 \
  --warmup \
```
```
# torch.compile 21.01s
sglang generate \
  --model-path=/model/qwen-image/ \
  --prompt='A fantasy landscape with mountains and a river, detailed, vibrant colors' \
  --width=1024 \
  --height=1024 \
  --vae-precision bf16 \
  --num-inference-steps=20 \
  --enable-torch-compile \
  --warmup \
```

910B, FLUX.1

```
# base: 16.91s
sglang generate \
  --model-path=/model/FLUX.1/ \
  --prompt='A fantasy landscape with mountains and a river, detailed, vibrant colors' \
  --width=1024 \
  --height=1024 \
  --vae-precision bf16 \
  --num-inference-steps=28 \
  --warmup \
```
```
# torch.compile 15.58s
sglang generate \
  --model-path=/model/FLUX.1/ \
  --prompt='A fantasy landscape with mountains and a river, detailed, vibrant colors' \
  --width=1024 \
  --height=1024 \
  --vae-precision bf16 \
  --num-inference-steps=28 \
  --enable-torch-compile \
  --warmup \
```

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review Process

1. Ping Merge Oncalls to start the PR flow. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - `/tag-run-ci-label`, `/rerun-failed-ci`, `/tag-and-rerun-ci`
4. After green CI and required approvals, ask Merge Oncalls to merge.

### CODE REVIEW

File: python/sglang/multimodal_gen/runtime/pipelines_core/stages/denoising.py [hunk 1]
```diff
@@ -137,16 +138,28 @@ def _maybe_enable_torch_compile(self, module: object) -> None:
             module, nn.Module
         ):
             return
-        try:
-            import torch._inductor.config as _inductor_cfg
+        compile_kwargs: dict[str, Any] = {"fullgraph": False, "dynamic": None}
```
Comment (zhuyijie88) [inline python/sglang/multimodal_gen/runtime/pipelines_core/stages/denoising.py:141]: how about _maybe_enable_torch_compile of MOVADenoisingStage in mova.py?

Comment (Alisehen) [reply python/sglang/multimodal_gen/runtime/pipelines_core/stages/denoising.py:141]: Thanks, I’ve aligned it with the denoising-stage change and now explicitly use the torchair backend on NPU there as well.

File: python/sglang/multimodal_gen/runtime/pipelines_core/stages/model_specific_stages/mova.py [hunk 2]
```diff
@@ -208,16 +209,31 @@ def _maybe_enable_torch_compile(self, module: nn.Module, server_args: ServerArgs
...
-            import torch._inductor.config as _inductor_cfg
-
-            _inductor_cfg.reorder_for_compute_comm_overlap = True
-        except ImportError:
-            pass
-        mode = os.environ.get("SGLANG_TORCH_COMPILE_MODE", "max-autotune-no-cudagraphs")
-        logger.info("Compiling %s with mode: %s", module.__class__.__name__, mode)
+        compile_kwargs: dict[str, object] = {"fullgraph": False, "dynamic": None}
+
+        if current_platform.is_npu():
```
Comment (mickqian) [inline python/sglang/multimodal_gen/runtime/pipelines_core/stages/model_specific_stages/mova.py:214]: please avoid this

Comment (Alisehen) [reply python/sglang/multimodal_gen/runtime/pipelines_core/stages/model_specific_stages/mova.py:214]: The platform split itself is hard to avoid here because the torch.compile configuration is inherently different on NPU vs non-NPU. We need an explicit backend selection on NPU, while other platforms should continue to use the existing inductor-mode  path.  I’m happy to move _maybe_enable_torch_compile to a shared helper, but some form of platform-specific branching is still required.

**Review by ping1jing2 — APPROVED** [review]
